src/textual/_widget_navigation.py

```diff
@@ -0,0 +1,139 @@
+"""
+Utilities to move index-based selections backward/forward.
+
+These utilities concern themselves with selections where not all options are available,
+otherwise it would be enough to increment/decrement the index and use the operator `%`
+to implement wrapping.
+"""
+
+from __future__ import annotations
+
+from functools import partial
+from typing import TYPE_CHECKING, Literal, Protocol, Sequence
+
+from typing_extensions import TypeAlias, TypeVar
+
+if TYPE_CHECKING:
+    from .widget import Widget
+
+
+class Disableable(Protocol):
+    """Non-widgets that have an enabled/disabled status."""
+
+    disabled: bool
+
+
+Direction: TypeAlias = Literal[-1, 1]
+"""Valid values to determine navigation direction.
+
+In a vertical setting, 1 points down and -1 points up.
+In a horizontal setting, 1 points right and -1 points left.
+"""
+_D = TypeVar("_D", bound=Disableable)
+_W = TypeVar("_W", bound="Widget")
+
+
+def distance(index: int, start: int, direction: Direction, wrap_at: int) -> int:
+    """Computes the distance going from `start` to `index` in the given direction
+
+    Starting at `start`, this is the number of steps you need to take in the given
+    `direction` to reach `index`, assuming there is wrapping at 0 and `wrap_at`.
+    This is also the smallest non-negative integer solution `d` to
+    `(start + d * direction) % wrap_at == index`.
+
+    The diagram below schematises the computation of `d1 = distance(2, 8, 1, 10)` and
```

Comment:
Nit-pick, but for consumption by the average reader of the documentation, "schematises" seems a little... unusual. I don't think I've ever seen the word before, and had to look it up to be sure exactly what it meant and that my inference was correct.

Perhaps something like "illustrates" would be more readable?

Reply:
Agreed.
The direct PT 🇵🇹 synonym is a fairly “simple” verb: esquematizar.